tests/v2/validation/auth/provider/provider_test.go

```diff
@@ -0,0 +1,484 @@
+
+	groupPrincipalID := newPrincipalID(authConfigID, "group", a.authConfig.Group, searchBase)
+	groupCRTB := &v3.ClusterRoleTemplateBinding{
+		ClusterID:        a.cluster.ID,
+		GroupPrincipalID: groupPrincipalID,
+		RoleTemplateID:   rbac.ClusterMember.String(),
+	}
+
+	logrus.Infof("Creating cluster role template binding for group [%v] with role [%v]", groupCRTB.GroupPrincipalID, groupCRTB.RoleTemplateID)
+
+	_, err = authAdmin.Management.ClusterRoleTemplateBinding.Create(groupCRTB)
+	require.NoError(a.T(), err, "Error occured while creating cluster role template binding")
+
+	defaultProject, err := projects.GetProjectByName(authAdmin, a.cluster.ID, "Default")
+	require.NoError(a.T(), err)
+
+	for _, v := range a.authConfig.NestedUsers {
+		nestedUserPrincipalID := newPrincipalID(authConfigID, "user", v.Username, searchBase)
+
+		userPRTB := &v3.ProjectRoleTemplateBinding{
+			ProjectID:        defaultProject.ID,
+			GroupPrincipalID: nestedUserPrincipalID,
+			RoleTemplateID:   rbac.ProjectOwner.String(),
+		}
+
+		logrus.Infof("Creating project role template binding for user [%v] with role [%v]", userPRTB.GroupPrincipalID, userPRTB.RoleTemplateID)
+		userPRTBResp, err := authAdmin.Management.ProjectRoleTemplateBinding.Create(userPRTB)
+		require.NoError(a.T(), err)
+
+		logrus.Infof("Verifying project role template binding has created for user [%v]", v.Username)
+		assert.NotNilf(a.T(), userPRTBResp, "Error occured while: project role template binding is created for user [%v]", v.Username)
+	}
+
+	var principalIDs []string
+
+	principalIDs = append(principalIDs, newPrincipalID(authConfigID, "group", a.authConfig.DoubleNestedGroup, searchBase))
+	for _, v := range a.authConfig.DoubleNestedUsers {
+		principalIDs = append(principalIDs, newPrincipalID(authConfigID, "user", v.Username, searchBase))
+	}
+
+	logrus.Infof("Verifying access mode updated to restrict access to only the authorized users & groups")
+
+	existing, updates := newWithAccessMode(a.T(), authAdmin, authConfigID, "required", principalIDs)
+	newAuthConfig, err := update(existing, updates)
+
+	require.NoError(a.T(), err)
+	assert.Equal(a.T(), existing.AccessMode, newAuthConfig.AccessMode, "Error occured while: access mode updated to restrict access to only the authorized users & groups")
+
+	for _, v := range a.authConfig.DoubleNestedUsers {
+		user := &v3.User{
+			Username: v.Username,
+			Password: v.Password,
+		}
+		_, err := login(authAdmin, authProvider, user)
+
+		logrus.Infof("Verifying logging as user [%v], should be able to login", v.Username)
+		assert.NoErrorf(a.T(), err, "Error occured while: logging as user [%v], should be able to login", v.Username)
+
+	}
+
+	for _, v := range slices.Concat(a.authConfig.Users, a.authConfig.NestedUsers) {
+		user := &v3.User{
+			Username: v.Username,
+			Password: v.Password,
+		}
+		_, err := login(authAdmin, authProvider, user)
+
+		logrus.Infof("Verifying logging as user [%v], should be able to login", v.Username)
```

Comment:
Shouldn't the users be not able to login as the access was set to `required` in line 342 and only `DoubleNestedGroup` were included in principalIDs from line 335 or am I getting it wrong?